src/util.js

```diff
@@ -6,11 +6,19 @@ function isInteger(n) {
   return n % 1 === 0;
 }
 
-var Null = t.irreducible('Null', function (x) { return x === null; });
-var Int = t.irreducible('Int', isInteger);
+function isNumber(n) {
+	return !isNaN(n) 
+}
 
+var Null = t.irreducible('Null', function (x) {
+  return x === null;
+});
+var Int = t.irreducible('Int', isInteger);
+var Num = t.irreducible('Num', isNumber)
```

Comment:
Is it possible to use JS's Number class to check if it's a number (and integer as well actually)?

Seems  like its interface supports these use cases and would really simplify this code.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number

Reply:
I think the method has to resolve to True or False, and for Integer, I think we would want to make sure it is a whole number, but I could make it `Number(n) !== NaN` for Number if that seems clearer?

Reply:
I meant the Number class built-in methods:
[Number.isInteger](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isInteger)
[Number.isNan](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isNaN)

They both return bools - so something like:

```javascript
var Int = t.irreducible('Int', Number.isInteger)
var Num = t.irreducible('Num', Number.isNan)
```

Reply:
I added the last bit to my comment after I tested those out. `Number.isNaN` works with strings or numbers, but `Number.isInteger('3')` results in `false`

Reply:
could do something like `Number.isInteger(parseInt(n))` for Integer

Reply:
Unfortunately - it goes back to failing when I sub out the method, even though in a node shell it lets me give a string to `Number.isNaN` in this use case it doesn't seem to work. So I don't think we can use it.

Reply:
Hummm, that's really weird. Sounds good though.